recovery/elliptics_recovery/dc_recovery.py

```diff
@@ -50,34 +50,38 @@ def remove(self):
                 log.info("Safe mode is turned on. Skip removing key: {0}".format(repr(self.key)))
             else:
                 log.info("Dry-run mode is turned on. Skip removing key: {0}.".format(repr(self.key)))
-            self.callback()
         else:
             try:
                 log.info("Removing key: {0} from group: {1}".format(self.key, self.session.groups))
                 remove_result = self.session.remove(self.key)
                 remove_result.connect(self.on_remove_corrupted)
+                return
             except:
                 log.exception("Failed to remove key: {0} from groups: {1}".format(self.key, self.session.groups))
-                self.callback()
+
+        self.on_complete()
 
     def on_remove_corrupted(self, results, error):
```

Comment:
Just to be clear, there is no more corruption flavor here, so let's make it just `on_remove`, ok?